Don't use the MSBuild server process for init-build.proj.

crummel opened the pull request:
This appears to fix an issue where our SdkResolver is not loaded because there are still lingering dotnet processes when we try to binplace the resolver DLL.

Fixes https://github.com/dotnet/source-build/issues/3021.

baronfel commented:
You _should_ also be able to shutdown any persistent servers (MSBuild or compiler) with `dotnet build-server shutdown` immediately prior to any command that needs access to any potentially-loaded files.

MichaelSimons commented:
@baronfel, Do you know if there was any behavior change in this space?  This code hasn't changed since the beginning of 7.0 and recently broke.  Is it expected to have to explicitly shutdown the build server to get new SdkResolvers to load?

baronfel commented:
Yeah - in 7.0 rc1 MSBuild server was enabled by default, which is why I think you're seeing this behavior now. From that time we've squashed a few bug in it, but we are still deciding if the feature should be on for the release. I'm not sure if I'd classify this specific interaction as a bug - for that I'll tag @rokonec for his opinion.

MichaelSimons commented:
@baronfel - Thanks for the explanation.

@crummel - Thoughts on using `dotnet build-server shutdown`?

crummel commented:
I think using the shutdown command is better as it's more clear in the intent.  I've changed the PR to use that and added a comment about what we're trying to do.

rokonec commented:
I would very like to understand why is  lingering dotnet processes of MSBuild server an issue for you? Can you please explain...

MichaelSimons commented:
@crummel - you mentioned the MSBuild server is going to be disabled by default for 7.0 and this change would not be required.  Can you link to the corresponding msbuild PR issue discussing this?

baronfel commented:
Saving some time since I saw this pop up: https://github.com/dotnet/sdk/pull/28369 is the PR to change the default behavior to be an opt-in instead of an opt out. we got tactics approval yesterday so should merge today.

MichaelSimons commented:
[rokonec](https://github.com/rokonec) - Can you answer the question if it is expected to have to explicitly shutdown the build server to get new SdkResolvers to load?  We are trying to make an informed decision if we should proceed with this PR even though the default behavior is being changed to opt-in?

rokonec commented:
@MichaelSimons I believe `shutdown servers` is safer approach. It will still work if in future server will be on by default.
Additionally it will shutdown all msbuild worker nodes processes which could also cause stale-cache-issues in some cases.
